File: Data Structures & Algorithms/buy-and-sell-crypto/submission-1.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int profit = 0;

        int l = 0;
        int r = 1;

        while (r < prices.size()) {
            profit = max(profit, prices[r] - prices[l]);
            if (prices[r] < prices[l]) {
                l = r;
            }
            r += 1;
        }
        return profit;
    }
};

File: Data Structures & Algorithms/max-water-container/submission-1.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        i, j = 0, len(heights)-1
        max_area = 0
        while (i < j):
            width = j-i
            height = min(heights[i], heights[j])
            area = width * height
            #print(i)
            #print(j)
            max_area = max(max_area, area)
            if (heights[i] < heights[j]):
                i+=1
            else:
                j-=1

        return max_area
        